tests/unit/utils/test_network.py

```diff
@@ -701,3 +701,15 @@ def test_generate_minion_id_with_long_hostname(self):
             # An exception is raised if unicode is passed to socket.getfqdn
             minion_id = network.generate_minion_id()
         assert minion_id != '', minion_id
+
+    def test_filter_by_network(self):
```

Comment:
In different tests, test the corner cases (empty, None). I would split the tests more, one for each case (dict, no dict for IPs,, None, empty list, list with matches, list without matches for network) Is more work, but each test will be like 2 lines and you will be super sure that all cases are covered, for whatever semantic you choose.

Reply:
I looked at the `test_is_ipv(4|6)` as a reference which do it like this, but I agree that it is better to split them up.

Reply:
Right. Do not use the current tests as a reference for good behavior : ( The plan is try to do the right thing.

For example, the new code is moving out of `unittest` and moving to `pytest`. This means that instead of `self.assertEqual(a, b)` the code will be `assert a == b`. But you will see tons of code using the first one (I still write it like that, to be honest, that is why I didn't complain in this code)